cryptographic_estimators/UOVEstimator/uov_estimator.py

```diff
@@ -0,0 +1,116 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from .uov_algorithm import UOVAlgorithm
+from .uov_problem import UOVProblem
+from ..base_estimator import BaseEstimator
+from math import inf
+
+
+class UOVEstimator(BaseEstimator):
+    """
+    Construct an instance of UOVEstimator
+
+    INPUT:
+
+    - ``n`` -- number of variables
+    - ``m`` -- number of polynomials
+    - ``q`` -- order of the finite field (default: None)
+    - ``w`` -- linear algebra constant (default: 2)
+    - ``theta`` -- exponent of the conversion factor (default: 2)
+        - If ``0 <= theta <= 2``, every multiplication in GF(q) is counted as `log2(q) ^ theta` binary operation.
+        - If ``theta = None``, every multiplication in GF(q) is counted as `2 * log2(q) ^ 2 + log2(q)` binary operation.
+    - ``h`` -- external hybridization parameter (default: 0)
+    - ``excluded_algorithms`` -- a list/tuple of MQ algorithms to be excluded (default: [Lokshtanov])
+    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
+    - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
+    - ``bit_complexities`` -- determines if complexity is given in bit operations or basic operations (default 1: in bit)
+
+    """
+    excluded_algorithms_by_default = []
```

Comment:
In the function comment you say that `Lokshtanov` is excluded. But here is nothing excluded.

Reply:
Done